### PR TITLE
Feat/options UI

### PR DESCRIPTION
**Pull request by beingsie**
## PR description:

This PR suggests **UI change**s to the options/settings page:
- Introduces the addition of [Font Awesome Icons](https://fontawesome.com/search?o=r&m=free&s=solid).
  - via [**CDN**](https://cdnjs.com/libraries/font-awesome)
- Added Google Font - **Inter** [Inter Regular](https://fontawesome.com/search?o=r&m=free&s=solid).
  - For a better displayed settings UI
    - via **CSS Import**

## Related Issues

No related issues but future issues can be created for further UI changes similar to the one included in this PR.

## Screenshots

Screenshot 2023-10-15 at 2 53 58 AM

## Additional Notes

Please note colors can be altered and new ones can be suggested. Original color scheme was kept to follow original developers design.

### CODE REVIEW

**Comment by beingsie** [comment]
This PR would close issue #16.

**Comment by prakhartiwari0** [comment]
Will Font awesome icons be available offline? @beingsie

**Comment by beingsie** [comment]
> Will Font awesome icons & Google Fonts be available offline? @beingsie

Thank you for bringing this up @prakhartiwari0. I have just pushed a commit that implements offline functionality for both fonts.

**Comment by prakhartiwari0** [comment]
Good to know that you have implemented offline support as well. I will take a look and see if it's working or not in my system.

Also, we should be having universal fonts, icons, so that they is no inconsistency.
That's why for this PR to be merged, we need to decide if these fonts, icons, etc. are something that will be used everywhere in the UI.

**Comment by prakhartiwari0** [comment]
And I should mention that the whole options page is just a rough implementation of clear data button that I needed during building this extension. But everything you see will be changed, like colours, elements, layout etc.

But because there is no parallel issue of improving options page, I will merge this PR and then raise issues for it. That ofcourse will be after taking some important decisions I talked about in the previous comment.

**Comment by beingsie** [comment]
Both of your messages I agree with. I was also thinking about the overall design and overall consistency of icons, fonts, colors etc.
- I would continue with my suggestion of using Font Awesome for the icons as they have a huge category of icons that are **free** to use for both personal and commercial products.
- I'm a UI designer and can further suggest an overall design if requested.

File: src/options/options.html [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
 <html lang="en">
+
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
     <!-- <link rel="stylesheet" href="../forExtensionPages.css">
     <link rel="stylesheet" href="./style.css"> -->
+    <!-- Font Awesome CDN -->
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/all.min.css"
```
Comment (prakhartiwari0) [inline src/options/options.html:11]: The delete icon is loading from here that's why it's not coming when I remove this line. I guess it should get this from the `.woff` files you have stored inside `font-awesome` folder in `assets`, right?

File: src/options/style.css [hunk 2]
```diff
@@ -1,10 +1,95 @@
...
+    src: url('../assets/inter/Inter-Regular.woff2') format('woff2'),
+        url('../assets/inter/Inter-Regular.woff') format('woff');
+    font-weight: normal;
+    font-style: normal;
+    font-display: swap;
+}
+
+/* Import Font Awesome */
+@font-face {
+    font-family: 'Font Awesome 6 Free Solid';
```
Comment (prakhartiwari0) [inline src/options/style.css:13]: This is not working on my side, the icon only comes when I keep the link code in the HTML

**Review by prakhartiwari0 — CHANGES_REQUESTED** [review]
Please solve this issue with Font Awesome icons. They are still loading from the link tag in the `options.html` and not from the `.woff` files.

**Comment by prakhartiwari0** [comment]
> * I would continue with my suggestion of using Font Awesome for the icons as they have a huge category of icons that are **free** to use for both personal and commercial products.

Sure then, we will use them.



> * I'm a UI designer and can further suggest an overall design if requested.
That's amazing, it will take some time for the basic layout to be created. When it's done, you can suggest and improve the UI :smile:

**Comment by prakhartiwari0** [comment]
I have tested the extension from my side, please take a look at the problems I have shared. And please correct me if I'm getting something wrong, as I haven't used font awesome before.

**Comment by beingsie** [comment]
> I have tested the extension from my side, please take a look at the problems I have shared. And please correct me if I'm getting something wrong, as I haven't used font awesome before.

Looking into this now thank you.

**Comment by beingsie** [comment]
I have pushed a commit that **fixes this issue**.
- Font Awesome requires a different method when using their font as apposed to the way regular web fonts work like Google Fonts where you can just import via the CSS only.

I also added 2 other icon types that come included in Font Awesome:
- Outlined
- Brands

This allows you to use `brand` type icons (Github etc.) and the outlined version of the current `solid` icon font.

Please review and let me know if you keep encountering the same issues.

**Comment by prakhartiwari0** [comment]
Does it Looks good?

**Comment by prakhartiwari0** [comment]
Is it looking good? @beingsie 

| Without Hover | On Hover |
| --- | --- |
| image | image |

**Comment by beingsie** [comment]
> Is it looking good? @beingsie
> 
> Without Hover	On Hover
> image	image

Hey great changes! Much better than the green color scheme from before, the buttons adhere well to this new color scheme. Great update!

**Comment by beingsie** [comment]
> It's working now @beingsie, good work. Let me make some colour changes and then merge this PR.
> 
> I wanted to discuss about the Font family. Is there a specific reason you chose [Inter](https://fonts.google.com/specimen/Inter)? Well, it doesn't actually matter why you chose that, but what fonts do you think will be suitable for this extension? Keep in mind that this extension needs to be accessible, easy to use and fast. So it means that we need to select fonts that are easy to read and are not bulky. (not a designer so ignore if make any mistake while talking about this stuff)
> 
> I'm creating a discussion where will be deciding what font to use, please share your recommendations there, and if possible, try and test different fonts everywhere in the Extension UI and share the Screenshot of how the Font is looking everywhere. I'll add these instructions to that Discussion too.

Inter was chosen out of no particular reason, it's a designers go-font for creating quick clean interfaces but any font that you suggest as clean and not bulky can work.

Since we don't have a final design I decided to use Inter as it's a commonly used font for it's simplicity and clean design.
- Once a final design has been chosen we can approach a final font.

**Comment by prakhartiwari0** [comment]
@beingsie I have created the Discussion, please take your time and share some good recommendations #25

**Comment by prakhartiwari0** [comment]
> * Once a final design has been chosen we can approach a final font.

Oh so you mean that we should first build the basic layout and overall design of the extension and then think about the Font Family?

**Pull request merged**